### PR TITLE
Fix #1952 - Implement iR <hashes>

### LINKED ISSUE #1952: iR <hashalgo>
**Is your feature request related to a problem? Please describe.**
To be able to hash resource the same way we can hash Section like `iS md5` -> `iR md5`.
Just like for iS it should show the hash for non-virtual

`iRj` algo should also work

**Describe the solution you'd like**
`iR algo`
`iRj algo` should also work

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fix #1952

### CODE REVIEW

**Comment by wargio** [comment]
@Maijin Let me know if this suits your need

**Comment by XVilka** [comment]
You will need to change Cutter too, but only after 2.0.4 release of it is out: 
- https://github.com/rizinorg/cutter/commit/896be642ffab4625c16e7af998b397e140bd6c02
- https://github.com/rizinorg/cutter/milestone/23

**Comment by wargio** [comment]
@XVilka why not now? or is more a future feature?

**Comment by XVilka** [comment]
I never said it's bad, just pointing out it should be done.

File: librz/core/cbin.c [hunk 1]
```diff
@@ -4961,42 +4959,112 @@ RZ_API bool rz_core_bin_memory_print(RzCore *core, RzBinFile *bf, RzCmdStateOutp
...
+				pj_ks(state->d.pj, hash, digest);
+			}
+		}
+		ht_pp_free(digests);
+	}
+end:
+	pj_end(state->d.pj);
+}
+
+RZ_API bool rz_core_bin_resources_print(RzCore *core, RzBinFile *bf, RzCmdStateOutput *state, RzList *hashes) {
```
Comment (XVilka) [inline librz/core/cbin.c]: Allow `hashes` to be NULL, add `RZ_NULLABLE`, also call `rz_list_foreach` only when hashes are not NULL.

File: librz/include/rz_core.h [hunk 2]
```diff
@@ -904,7 +904,7 @@ RZ_API bool rz_core_bin_fields_print(RZ_NONNULL RzCore *core, RZ_NONNULL RzBinFi
 RZ_API bool rz_core_bin_headers_print(RZ_NONNULL RzCore *core, RZ_NONNULL RzBinFile *bf);
 RZ_API bool rz_core_bin_dwarf_print(RZ_NONNULL RzCore *core, RZ_NONNULL RzBinFile *bf, RZ_NONNULL RzCmdStateOutput *state);
 RZ_API bool rz_core_bin_memory_print(RZ_NONNULL RzCore *core, RZ_NONNULL RzBinFile *bf, RZ_NONNULL RzCmdStateOutput *state);
-RZ_API bool rz_core_bin_resources_print(RZ_NONNULL RzCore *core, RZ_NONNULL RzBinFile *bf, RZ_NONNULL RzCmdStateOutput *state);
+RZ_API bool rz_core_bin_resources_print(RZ_NONNULL RzCore *core, RZ_NONNULL RzBinFile *bf, RZ_NONNULL RzCmdStateOutput *state, RzList *hashes);
```
Comment (XVilka) [inline librz/include/rz_core.h]: Add `RZ_NULLABLE` to `hashes`.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/core/cbin.c (hunk 1 above)
Comment (wargio) [reply librz/core/cbin.c]: actually
```c
#define rz_list_foreach(list, it, pos) \
	if (list) \
		for (it = list->head; it && (pos = it->data, 1); it = it->n)
```

**Review by XVilka — APPROVED** [review]

**Pull request merged**